docker.go

```diff
@@ -343,6 +364,26 @@ func (c *DockerContainer) Terminate(ctx context.Context) error {
 	c.sessionID = ""
 	c.isRunning = false
 
+	// Remove additional volumes if any.
+	// TODO: simplify this when when perform the client refactor.
+	if len(defaultOptions.Volumes) == 0 {
+		return nil
+	}
+
+	client, err := NewDockerClientWithOpts(ctx)
+	if err != nil {
+		return fmt.Errorf("docker client: %w", err)
+	}
+
+	defer client.Close()
+
+	// Best effort to remove all volumes.
+	for _, volume := range defaultOptions.Volumes {
+		if errRemove := client.VolumeRemove(ctx, volume, true); errRemove != nil {
+			errs = append(errs, fmt.Errorf("volume remove %q: %w", volume, errRemove))
+		}
+	}
+
```

Comment:
suggest: leverage the Cleanup method.
```suggestion
	if err := options.Cleanup(ctx); err != nil {
		errs = append(errs, err)
	}
```